server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1924,6 +1935,23 @@ public Volume migrateVolume(MigrateVolumeCmd cmd) {
                                 throw new InvalidParameterValueException("Cannot migrate ROOT volume of a stopped VM to a storage pool in a different VMware datacenter");
                             }
                         }
+
```

Comment:
I disagree with you here. Those huge (really huge) methods that ACS has are the rock in our shoes when writing unit and integration tests (not talking about functional tests here). It is almost impossible to write and maintain a test case for a method with 50, 100, 500+ lines. 

Even though a method is not used in more than one place, I believe it is a good practice to break them into smaller ones whenever possible. That facilitates the writing of unit and integration tests cases.